torchvision/prototype/datasets/_builtin/usps.py

```diff
@@ -0,0 +1,72 @@
+import bz2
+import functools
+from typing import Any, Dict, List, Tuple, BinaryIO, Iterator
+
+import numpy as np
+import torch
+from torchdata.datapipes.iter import IterDataPipe, IterableWrapper, LineReader, Mapper
+from torchvision.prototype.datasets.utils import Dataset, DatasetInfo, DatasetConfig, OnlineResource, HttpResource
+from torchvision.prototype.datasets.utils._internal import hint_sharding, hint_shuffling
+from torchvision.prototype.features import Image, Label
+
+
+class USPSFileReader(IterDataPipe[torch.Tensor]):
```

Comment:
Could you refactor this now that your PR (thank you!) pytorch/data#312 was merged? My guess is that all the processing you do to the `line` should go into `_prepare_sample`.